### ISSUE TITLE
Specify container port protocol for ApplicationLoadBalancedFargateService

### ISSUE THREAD

**Issue by jdavisp3** [aws/aws-cdk#7266]
Specify target group protocol when creating an application load balanced service.

### Use Case

To use the ECS Pattern for a `LoadBalancedFargateService` when the container serves HTTPS.

### Proposed Solution

Make the target protocol an option in the constructor.

---

This is a :rocket: Feature Request

**Comment by bvtujo** [comment]
Hi jdavisp3, just to be clear, are you describing the situation where you need the routing capabilities of the ALB, but need a new HTTPS connection to be originated at the load balancer and routed to your container targets?

**Comment by jdavisp3** [comment]
> Hi jdavisp3, just to be clear, are you describing the situation where you need the routing capabilities of the ALB, but need a new HTTPS connection to be originated at the load balancer and routed to your container targets?

That's exactly right.

**Comment by bvtujo** [comment]
Thanks for the reply! This fix seems pretty quick at the moment, so barring any unexpected complications we should have this out as an optional parameter shortly.

**Comment by jdavisp3** [comment]
> Thanks for the reply! This fix seems pretty quick at the moment, so barring any unexpected complications we should have this out as an optional parameter shortly.

Awesome! I was able to convince the current version to do this, but I had to add a lot of extra code so having this as an optional parameter will be 👌

**Comment by bvtujo** [comment]
Would you be able to share any (anonymized, of course) snippets of what you did to coerce the current implementation to behave properly? It would help us understand some of the ways our customers are innovating using the CDK to meet their own needs.

**Comment by jdavisp3** [comment]
Sure so first I added two port mappings to the container:

```
container.add_port_mappings(
            ecs.PortMapping(
                container_port=8000, protocol=elasticloadbalancingv2.ApplicationProtocol.HTTPS
            )
        )
```

Making sure HTTPS came after 80 (I don't need 80 but that's the one that the construct will use so I can configure HTTPS myself).

Then after creating the whole service I did this:

```
        # so the target group for 80 is healthy even though not really used
        load_balanced_service.target_group.configure_health_check(
            path="/health", port="8000", protocol=elasticloadbalancingv2.ApplicationProtocol.HTTPS
        )

        fargate_service = load_balanced_service.service

        listener = load_balancer.add_listener(
            "https-listener",
            certificates=[certificate],
            protocol=elasticloadbalancingv2.ApplicationProtocol.HTTPS,
        )

        target = fargate_service.load_balancer_target(
            container_name="app-container",
            container_port=8000,
            protocol=elasticloadbalancingv2.ApplicationProtocol.HTTPS,
        )

        target_group = listener.add_targets(
            "target-group",
            protocol=elasticloadbalancingv2.ApplicationProtocol.HTTPS,
            targets=[target],
        )

        target_group.configure_health_check(
            path="/health", protocol=elasticloadbalancingv2.ApplicationProtocol.HTTPS
        )
```

**Comment by bvtujo** [comment]
Related #6428

**Comment by jdavisp3** [comment]
To shed some additional light on the question of "why do you need HTTPS if it's inside your VPC" I would mention the HIPAA guidelines of "encrypt at rest, encrypt in transit". Not to suggest that the networking inside AWS inside isn't rock-solid already, but in general it's easier to satisfy regulatory requirements by following them even when it's not strictly technically necessary.

@kohidave

**Comment by jtwolgamott** [comment]
@SomayaB I have the same requirement as @jdavisp3. Building systems that fall under the HIPAA guidelines for handling of PHI. Unless AWS can provide guidance/documentation indicating that traffic within a VPC is protected in accordance with HIPAA guidelines, I'm required to ensure end-to-end encryption.

**Comment by jtwolgamott** [comment]
Related https://github.com/aws/aws-cdk/issues/7989

**Comment by hoegertn** [comment]
Also #11381

**Comment by bvtujo** [comment]
This is resolved as of merge of #11381
